cpp/src/fil/internal.cuh

```diff
@@ -266,6 +302,90 @@ struct forest_params_t {
 /// FIL_TPB is the number of threads per block to use with FIL kernels
 const int FIL_TPB = 256;
 
+static const uint32_t max_precise_int_float = 1 << 24;  // 16'777'216
+
+struct cat_feature {
```

Comment:
Does this structure store data associated with a categorical feature, e.g. the maximum matching category value or the number of nodes with this feature?

In that case, I would use a different name for this structure, e.g. `categorical_properties` or something like that.